tests/test_FBbasis2D.py

```diff
@@ -2,330 +2,80 @@
 from unittest import TestCase
 
 import numpy as np
+from parameterized import parameterized_class
 from pytest import raises
+from scipy.special import jv
 
 from aspire.basis import FBBasis2D
-from aspire.image import Image
-from aspire.utils import complex_type, real_type, utest_tolerance
+from aspire.utils import complex_type, real_type
+from aspire.utils.coor_trans import grid_2d
+from aspire.utils.random import randn
+
+from ._basis_util import Steerable2DMixin
 
 DATA_DIR = os.path.join(os.path.dirname(__file__), "saved_test_data")
 
 
-class FBBasis2DTestCase(TestCase):
+# NOTE: Class with default values is already present, so don't list it below.
+@parameterized_class(
+    ("L", "dtype"),
+    [
+        (8, np.float64),
+        (16, np.float32),
+        (16, np.float64),
+        (32, np.float32),
+        (32, np.float64),
+    ],
+)
+class FBBasis2DTestCase(TestCase, Steerable2DMixin):
+    L = 8
+    dtype = np.float32
+
     def setUp(self):
-        self.dtype = np.float32
-        self.basis = FBBasis2D((8, 8), dtype=self.dtype)
+        self.basis = FBBasis2D((self.L, self.L), dtype=self.dtype)
+        self.seed = 9161341
 
     def tearDown(self):
         pass
 
-    def testFBBasis2DIndices(self):
+    def testElement(self):
+        ell = 1
+        sgn = -1
+        k = 2
+
         indices = self.basis.indices()
+        ells = indices["ells"]
+        sgns = indices["sgns"]
+        ks = indices["ks"]
+
+        g2d = grid_2d(self.L, dtype=self.dtype)
+        mask = g2d["r"] < 1
+
+        r0 = self.basis.r0[k, ell]
+
+        im = np.zeros((self.L, self.L), dtype=self.dtype)
+        im[mask] = jv(ell, g2d["r"][mask] * r0)
+        im *= np.sqrt(2**2 / self.L**2)
```

Comment:
We want to approximate the integral using a sum over pixels. Each pixel has area `2 ** 2 / self.L ** 2` and we want to normalize so that the L² norm is one, hence the square root.

Reply:
Ah ok. Makes sense. Thanks for the clarification.